src/App.tsx

```diff
@@ -1,14 +1,65 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, {
+  useCallback, useEffect, useMemo, useState,
+} from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos, getUser } from './api';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [currentTodo, setCurrentTodo] = useState<Todo | undefined>();
+  const [currentUser, setCurrentUser] = useState<User | undefined>();
+  const [query, setQuery] = useState('');
+  const [filterType, setFilterType] = useState('all');
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos);
+  }, []);
+
+  const filteredTodos = useMemo(() => {
+    let currentTodos = todos;
```

Comment:
you don't need to reassign `todos` to a new variable. You only use `filter` on this that don't modify this array. Operate on `todos` when you filter, and create new variable to assign the result to it

Reply:
I do not think I understand what you are saying; you say not to reassign to a new variable, then you say to create a new variable to assign the result to it... when it's already exactly what I'm doing. The variable exists because there are 2 separate filters below, filter by completion status and by query, as it would be difficult to filter it using a single filter expression.